### PR TITLE
Authentication logic refactor

### LINKED ISSUE #337: Problems for user when loading app
User (Oskar P.) has problems with accessing vote.bcc.no. When he tries to reach main page all he sees is error: "
image
"
**Additional Description**
....

**Acceptance Criteria**
1. Problem is gone.

### PR DESCRIPTION
**Pull request by JakubC-projects**
Changes the authentication logic to match what was used in members.
Removes feathers token, so we can just use auth0 token.

I've also run the linter because I was sick of the linting errors/warning

closes \#337

### CODE REVIEW

File: server/src/app.hooks.ts [hunk 1]
```diff
@@ -220,7 +220,7 @@ export default {
     },
 
     error: {
-        all: [trackErrors,purgeErrors, logErrors],
+        all: [trackErrors,purgeErrors],
```
Comment (elpenny) [inline server/src/app.hooks.ts]: missing logErrror function call

File: server/src/services/authentication/authentication-helpers.ts [hunk 2]
```diff
@@ -17,4 +21,45 @@ export function getActiveRole(userRoles:Array<UserRole>):RoleName{
...
+export async function getUserBasedOnPayLoad(payload: Record<string, any>, app: any): Promise<User | null> {
+    const personID = payload['https://login.bcc.no/claims/personId'];
+    const user: User = await app.services.person.get(personID.toString());
+    user.authTime = payload.iat;
+    user.roles = user.related.roles;
+
+    user.activeRole = getActiveRole(user.related.roles);
+
+    return user;
 }
```
Comment (elpenny) [inline server/src/services/authentication/authentication-helpers.ts]: EOF empty line is missing

**Review by elpenny — APPROVED** [review]
Some changes are needed but it looks ok. Pity we dont have sufficient tests here.

File: server/src/app.hooks.ts (hunk 1 above)
Comment (JakubC-projects) [reply server/src/app.hooks.ts]: fixed

File: server/src/services/authentication/authentication-helpers.ts (hunk 2 above)
Comment (JakubC-projects) [reply server/src/services/authentication/authentication-helpers.ts]: fixed

**Pull request merged**